nessie/lakectl_util_test.go

```diff
@@ -0,0 +1,233 @@
+		t.Fatal("Failed to read ", goldenFile, err)
+	}
+	expected := sanitize(string(content), vars)
+	runCmdAndVerifyResult(t, cmd, expectFail, isTerminal, expected, vars)
+}
+
+func runCmdAndVerifySuccess(t *testing.T, cmd string, isTerminal bool, expected string, vars map[string]string) {
+	runCmdAndVerifyResult(t, cmd, false, isTerminal, expected, vars)
+}
+
+func runCmdAndVerifyFailure(t *testing.T, cmd string, isTerminal bool, expected string, vars map[string]string) {
+	runCmdAndVerifyResult(t, cmd, true, isTerminal, expected, vars)
+}
+
+func runCmdAndVerifyResult(t *testing.T, cmd string, expectFail bool, isTerminal bool, expected string, vars map[string]string) {
+	expanded, err := expandVariables(expected, vars)
+	if err != nil {
+		t.Fatal("Failed to extract variables for", cmd)
+	}
+	result, err := runShellCommand(cmd, isTerminal)
+	if expectFail {
+		require.Error(t, err, "Expected error in %s command did not occur. Output -%s", cmd, string(result))
+	} else {
+		require.NoError(t, err, "Failed to run %s command - %s", cmd, string(result))
+	}
+	require.Equal(t, expanded, sanitize(string(result), vars), "Unexpected output for %s command", cmd)
+}
+
+var (
+	timeStampRegexp = regexp.MustCompile(`timestamp: \d+\r?\n`)
+	timeRegexp      = regexp.MustCompile(`\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2} [\+|-]\d{4} \w{1,3}`)
+	commitIdRegExp  = regexp.MustCompile(`[\d|a-f]{64}`)
+)
+
+func removeProgramTimestamp(output string) string {
+	s := timeStampRegexp.ReplaceAll([]byte(output), []byte("timestamp: <TIMESTAMP>"))
+	s = timeRegexp.ReplaceAll(s, []byte("<DATE> <TIME> <TZ>"))
+	return string(s)
+}
+
+func removeRandomObjectKey(output string, objectPrefix string) string {
+	objectKeyRegExp := regexp.MustCompile(objectPrefix + `/[\d|a-f]{32}`)
+	s := objectKeyRegExp.ReplaceAll([]byte(output), []byte(objectPrefix+"/<OBJECT_KEY>"))
+	return string(s)
+}
+
+func removeCommitID(output string) string {
+	s := commitIdRegExp.ReplaceAll([]byte(output), []byte("<COMMIT_ID>"))
+	return string(s)
+}
+
+//
+// Utilities tests
+//
+func TestRunShellCommand(t *testing.T) {
+	checkTerminal := "if [ -t 1 ] ; then echo -n terminal; else echo -n pipe; fi"
+
+	output, err := runShellCommand(checkTerminal, true)
+	require.NoError(t, err, "Failed to run shell command in terminal context")
+	require.Equal(t, "terminal", string(output), "terminal was not detected properly")
+
+	output, err = runShellCommand(checkTerminal, false)
+	require.NoError(t, err, "Failed to run shell command out of terminal context")
+	require.Equal(t, "pipe", string(output), "pipe was not detected properly")
+}
+
+func TestExpandVariables(t *testing.T) {
+	s := "This is a string with ${number} ${vars} to expand. This $part $should ${not $expand"
+	expected := "This is a string with 2 elements to expand. This $part $should ${not $expand"
+	varMap := make(map[string]string)
+	expanded, err := expandVariables(s, varMap)
+	require.Error(t, err, "Expected error for empty mapping did not happen")
```

Comment:
Specifically here?
This is a test I used to debug the expandedVariables func, so I made it a bit more robust so it 'protects' the function. I believe expandVariables can only return 'var not found' error

Reply:
Sorry, if it's only one error then why not check that it is that expected error using `if !errors.Is(...)`?

Reply:
I tried and it seems not trivial.
erros.Is(...) compares the errors themselves and not just the messages, and since this is just an internal error that is created on the fly, I don't see how to do that easily, and since this is just a protection on an internal testing utility function, so I don't think it is worth the effort.
WDYT?

Reply:
Done